tensorflow_probability/python/positive_semidefinite_kernels/polynomial.py

```diff
@@ -142,26 +161,32 @@ def slope_variance(self):
     """Variance on slope parameter."""
     return self._slope_variance
 
+  @property
+  def intercept(self):
+    """Intercept of linear function that is exponentiated."""
+    return self._intercept
+
   @property
   def exponent(self):
     """Exponent of the polynomial term."""
     return self._exponent
 
   def _batch_shape(self):
-    return tf.broadcast_static_shape(
-        tf.broadcast_static_shape(
-            _maybe_shape_static(self.slope_variance),
-            _maybe_shape_static(self.bias_variance)),
-        _maybe_shape_static(self.exponent))
+    return reduce(
+      tf.broadcast_static_shape,
+      map(_maybe_shape_static, [self.slope_variance, self.bias_variance,
+                                self.intercept, self.exponent]))
 
   def _batch_shape_tensor(self):
-    return tf.broadcast_dynamic_shape(
-        tf.broadcast_dynamic_shape(
-            _maybe_shape_dynamic(self.slope_variance),
-            _maybe_shape_dynamic(self.bias_variance)),
-        _maybe_shape_dynamic(self.exponent))
+    return reduce(
+      tf.broadcast_dynamic_shape,
+      map(_maybe_shape_dynamic, [self.slope_variance, self.bias_variance,
+                                 self.intercept, self.exponent]))
 
   def _apply(self, x1, x2, param_expansion_ndims=0):
+    if self.intercept is not None:
+      x1 = x1 - self.intercept
```

Comment:
nit: maybe just inline these below; feels weird to re-interpret the meanings of `x1` and `x2`